synapse/rest/media/v1/thumbnailer.py

```diff
@@ -30,6 +30,26 @@ class Thumbnailer(object):
 
     def __init__(self, input_path):
         self.image = Image.open(input_path)
+        try:
+            # We don't use ImageOps.exif_transpose since it crashes with big EXIF
+            image_exif = self.image._getexif()
+            image_orientation = image_exif.get(0x0112)
+            transpose_method = {
+                2: Image.FLIP_LEFT_RIGHT,
+                3: Image.ROTATE_180,
+                4: Image.FLIP_TOP_BOTTOM,
+                5: Image.TRANSPOSE,
+                6: Image.ROTATE_270,
+                7: Image.TRANSVERSE,
+                8: Image.ROTATE_90
+            }.get(image_orientation)
+            if transpose_method is not None:
+                self.image = self.image.transpose(transpose_method)
+                # We don't need EXIF any more
+                self.image.info["exif"] = None
+        except Exception as e:
+            # A lot of parsing errors can happen when parsing EXIF
+            logger.debug("Error parsing image EXIF information: %s", e)
```

Comment:
1. I suggest making this at least an `info`, since most people have `debug` turned off.
2. We won't just reach this point for errors parsing the EXIF, but also when doing the transpose. We need to either make the `try` clause cover less code, or fix the log message (and comment) to better reflect the operations that might have caused the exception.